File: sdk/nodejs/cmd/pulumi-language-nodejs/testdata/published/tsnode/projects/l3-component-config-objects/primitiveComponent.ts
import * as pulumi from "@pulumi/pulumi";
import * as primitive from "@pulumi/primitive";

interface PrimitiveComponentArgs {
    numberArray: pulumi.Input<number[]>,
    booleanMap: pulumi.Input<Record<string, pulumi.Input<boolean>>>,
}

export class PrimitiveComponent extends pulumi.ComponentResource {
    constructor(name: string, args: PrimitiveComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        super("components:index:PrimitiveComponent", name, args, opts);
        const res = new primitive.Resource(`${name}-res`, {
            boolean: true,
            float: 3.5,
            integer: 3,
            string: "plain",
            numberArray: args.numberArray,
            booleanMap: args.booleanMap,
        }, {
            parent: this,
        });

        this.registerOutputs();
    }
}

File: sdk/python/cmd/pulumi-language-python/testdata/setuppy/local/projects/l3-component-config-objects/primitiveComponent.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import pulumi_primitive as primitive

class PrimitiveComponentArgs(TypedDict, total=False):
    numberArray: Input[list[float]]
    booleanMap: Input[Dict[str, bool]]

class PrimitiveComponent(pulumi.ComponentResource):
    def __init__(self, name: str, args: PrimitiveComponentArgs, opts:Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:PrimitiveComponent", name, args, opts)

        res = primitive.Resource(f"{name}-res",
            boolean=True,
            float=3.5,
            integer=3,
            string="plain",
            number_array=args["numberArray"],
            boolean_map=args["booleanMap"],
            opts = pulumi.ResourceOptions(parent=self))

        self.register_outputs({})
